### PR TITLE
Add tests for LDF parsing and serialization.  Cleanup LDF header

### PR DESCRIPTION
**Pull request by EmosewaMC**
- Implement a speedup by using overloaded operators to put non-strings directly to a stringstream as opposed to doing to_string first.
- Replace header guard with a traditional C style one
- Fix ReadFromString so it can handle a string of colons (rare case, but an edge case regardless)\
- Add further edge case testing for LDF keys to ensure bad keys are not parsed and to ensure correct behavior

All new tests are attached and were created _before_ making any changes and will be run through regression when the PR is finished

### CODE REVIEW

**Comment by EmosewaMC** [comment]
All tests pass after the fixes and most tests pass pre-fixes (except those which threw originally or had legitimate bugs.  Will now do a final sweep for optimization to prevent copying of so many strings

File: dMasterServer/MasterServer.cpp [hunk 1]
```diff
@@ -334,6 +334,15 @@ int main(int argc, char** argv) {
...
 		Game::im->GetInstance(1000, false, 0);
+		Game::im->GetInstance(1100, false, 0);
+		Game::im->GetInstance(1200, false, 0);
+		Game::im->GetInstance(1300, false, 0);
+		Game::im->GetInstance(1400, false, 0);
+		Game::im->GetInstance(1600, false, 0);
+		Game::im->GetInstance(1700, false, 0);
+		Game::im->GetInstance(1800, false, 0);
+		Game::im->GetInstance(1900, false, 0);
+		Game::im->GetInstance(2000, false, 0);
```
Comment (Wincent01) [inline dMasterServer/MasterServer.cpp:345]: This shouldn't be here.

**Review by Wincent01 — CHANGES_REQUESTED** [review]

File: dMasterServer/MasterServer.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dMasterServer/MasterServer.cpp:345]: Lol I forgot about this

**Review by aronwk-aaron — APPROVED** [review]

**Review by DarwinAnim8or — APPROVED** [review]
